### PR TITLE
feat(form): allow to prevent form submitting on hitting enter

### PR DESCRIPTION
**Pull request by rosendi**
Usage:

```
const SomeForm = Form.extend({
  defaults: {
    preventSubmitOnEnter: true
  },
  
  ...
});

### CODE REVIEW

File: src/blocks/form/form.js [hunk 1]
```diff
@@ -5,12 +5,17 @@ define(function(require) {
...
+            keypress: '_interceptEnterPress',
+
             submit: function(e) {
                 e.preventDefault();
 
                 this.submit();
             }
         },
+
+        preventEnterSubmit: false,
```
Comment (rosendi) [inline src/blocks/form/form.js]: Maybe we should call it `preventSubmitOnEnter`?

Comment (zouxuoz) [inline src/blocks/form/form.js]: move it to `defaults`

File: src/blocks/form/form.js [hunk 2]
```diff
@@ -245,6 +250,13 @@ define(function(require) {
                 .find('button')
                 .click();
         },
+
+        _interceptEnterPress: function(e) {
+            if (e.target.tagName.toUpperCase() !== 'TEXTAREA' && this.preventEnterSubmit && e.keyCode === 13) {
```
Comment (zouxuoz) [inline src/blocks/form/form.js]: swap first and second condition

File: src/blocks/form/form.js (hunk 1 above)
Comment (rosendi) [inline src/blocks/form/form.js]: @zouxuoz done

File: src/blocks/form/form.js (hunk 2 above)
Comment (rosendi) [inline src/blocks/form/form.js]: @zouxuoz done

**Comment by zouxuoz** [comment]
lgtm 👍

**Pull request merged**